packages/platform-core/src/errors/amplify_error.ts

```diff
@@ -121,3 +139,37 @@ export type AmplifyUserErrorOptions = Omit<
   AmplifyErrorOptions,
   'resolution'
 > & { resolution: string };
+
+const errorSerializer = (_: unknown, value: unknown) => {
+  if (value instanceof Error && value?.constructor.name === 'Error') {
+    return ErrorSerDe.serialize(value);
+  }
+  return value;
+};
+class ErrorSerDe {
```

Comment:
mostly preference, but it seems like these methods could just be functions in the module

Reply:
Yes. It would also remove `SerDe` which I find a bit confusing.
I'd just call it `ErrorSerializerDeserializer` if we want to keep it.

Reply:
Updated to `ErrorSerializerDeserializer` I like keeping the three functions together 🤷